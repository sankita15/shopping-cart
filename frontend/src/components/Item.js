import React from 'react';
import { Card, CardImg, CardBody, CardTitle, CardText, CardSubtitle } from 'reactstrap';
import * as PropTypes from 'prop-types';

const Item = ({ imageUrl, description, price, productName, id }) => (
    <a href={`/products/${id}`}>
        <div className="item-page">
            <Card className="item-card">
                <CardImg top width="100%" className="item-page-image" src={imageUrl} />
                <CardBody className="item-body">
                    <CardTitle>{productName}</CardTitle>
                    <CardSubtitle>
                                Rs.
                        {price}
                    </CardSubtitle>
                    <CardText>{description}</CardText>
                </CardBody>
            </Card>
        </div>
    </a>
);

Item.propTypes = {
    id: PropTypes.string.isRequired,
    productName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    imageUrl: PropTypes.string.isRequired,
};

export default Item;
