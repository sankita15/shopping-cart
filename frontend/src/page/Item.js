import React from 'react';
import {Card, CardImg, CardBody, CardTitle, CardText, CardSubtitle, Button, CardDeck} from "reactstrap";
import PropTypes from 'prop-types';

export default class Item extends React.Component {

    render() {
        const {item} = this.props;
        return (
            <div className="item-page">
                    <Card className="item-card">
                        <CardImg top width="100%" className="item-image" src={item.imageUrl}/>
                        <CardBody className="item-body">
                            <CardTitle>{item.productName}</CardTitle>
                            <CardSubtitle>Rs.{item.price}</CardSubtitle>
                            <CardText>{item.description}</CardText>
                            <Button className="btn btn-primary">View Details</Button>
                        </CardBody>
                    </Card>
            </div>
        )
    }
}

Item.propTypes = {
    item: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string,
        productName: PropTypes.string,
        productCode: PropTypes.string,
        releaseDate: PropTypes.number,
        lastModified: PropTypes.number,
        description: PropTypes.string,
        starRating: PropTypes.number,
        price: PropTypes.number,
        imageUrl: PropTypes.string,
        stock: PropTypes.number
    }))
};
