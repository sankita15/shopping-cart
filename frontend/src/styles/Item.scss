.item-page {

  width: 25%;
  height: 341px;
  float: left;

  .item-card {
    margin: 10px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    transition: 0.3s;

    .item-page-image {
      width: 200px;
      height: 200px;
      margin-left: 70px;
    }

    .item-body{
      color: black;
    }

    .btn-secondary {
      background-color: #007bff;
      border-color: #007bff;
    }

    .btn .btn-primary .btn-secondary {
      width: 100%;
      display: block;
    }

  }
}
