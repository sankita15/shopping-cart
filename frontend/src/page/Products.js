import React from 'react';
import Item from '../components/Item';
import Header from './Header';


export default class Products extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            products: [],
        };
    }

    componentDidMount() {
        fetch('/api/products', {
            credentials: 'include',
            headers: {
                'Content-Type': 'application/json',
            },
        })
            .then(res => (res.ok ? res.json() : Promise.reject(res.status)))
            .then(products => this.setState({ products }))
            .catch(e => console.warn(e));
    }

    render() {
        const { products } = this.state;

        return (
            <div className="product-login">
                <Header />
                {products.map(({ productName, price, description, imageUrl, id }) => (
                    <Item
                        productName={productName}
                        price={price}
                        description={description}
                        imageUrl={imageUrl}
                        id={id}
                        key={id}
                    />
                ))}
            </div>
        );
    }
}
