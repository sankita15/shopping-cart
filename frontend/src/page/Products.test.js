import React from 'react';
import { shallow } from 'enzyme';
import Products from './Products';
import Item from '../components/Item';


describe('Product page', () => {
    const createWrapper = () => shallow(<Products />);

    const products = [
        {
            'id': '5c61496906057ea38af387cd',
            'productName': 'Leaf Rake',
            'productCode': 'GDN-0011',
            'releaseDate': 1489916755570,
            'lastModified': 1549879657904,
            'description': 'Leaf rake with 48-inch wooden handle.',
            'price': 1995,
            'starRating': 3.2,
            'imageUrl': 'http://openclipart.org/image/300px/svg_to_png/26215/Anonymous_Leaf_Rake.png',
            'stock': 400,
        },
        {
            'id': '5c61496906057ea38af387ce',
            'productName': 'Garden Cart',
            'productCode': 'GDN-0023',
            'releaseDate': 1489805155570,
            'lastModified': 1549879657912,
            'description': '15 gallon capacity rolling garden cart',
            'price': 3295,
            'starRating': 4.2,
            'imageUrl': 'http://openclipart.org/image/300px/svg_to_png/58471/garden_cart.png',
            'stock': 20,
        },
    ];

    afterEach(() => {
        jest.clearAllMocks();
        fetch.resetMocks();
    });

    it('should match snapshot of Product page', async () => {
        fetch.mockResolvedValueOnce({ json: () => Promise.resolve([]), ok: true });

        const component = createWrapper();

        // eslint-disable-next-line no-undef
        await flushPromises();

        expect(component).toMatchSnapshot();
    });

    it('should render list of products', async () => {
        fetch.mockResolvedValueOnce({ json: () => Promise.resolve(products), ok: true });
        const component = createWrapper();

        expect(component.find(Item).length).toBe(0);

        expect(fetch).toHaveBeenCalledWith('/api/products', {
            credentials: 'include',
            headers: {
                'Content-Type': 'application/json',
            },
        });

        // eslint-disable-next-line no-undef
        await flushPromises();

        expect(component.find(Item).length).toBe(2);
    });
});
