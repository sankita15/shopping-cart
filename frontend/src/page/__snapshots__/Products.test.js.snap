// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page should match snapshot of Product page 1`] = `
<div
  className="product-login"
>
  <Header />
</div>
`;
